### PR TITLE
Elaborate on the database document

### LINKED ISSUE #131: Elaborating database description
The current database description contains no more than listing of attributes. We need to elaborate on why these tables and attributes are needed.

### PR DESCRIPTION
**Pull request by Huy-Ngo**
This resolve #131 

It describes more in details why the data are chosen to be stored and also add a graph to represent the relations between the entities.

Note that this is very rough since I'm not very good at writing.

### CODE REVIEW

File: docs/source/meth/database/index.rst [hunk 1]
```diff
@@ -1,19 +1,53 @@
...
+should be kept separate for comparision or combination.
+
+Therefore, there should be two one-to-many relations
+between ``Project`` and ``File``.
+
+However, a file in ``File`` table
+does not necessarily tie to any project, since they can be referred in comments
+or by academic assistants in their statistical reports.
+
+Visually, that can be represented in :numref:`rel`.
```
Comment (McSinyx) [inline docs/source/meth/database/index.rst:43]: To me it's ambiguous what *that* means in this context.

File: docs/source/meth/database/comment.rst [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
    The user who created the comment
 
 ``content`` : ``string``
    The content of the comment
 
+Since the discussion follows a forest structure, each comment recursively
+contains a list of replying comment.
+
 ``comments`` : ``array`` of ``Comment``
    List of ``Comment``\s replying to it
```
Comment (McSinyx) [inline docs/source/meth/database/comment.rst:17]: At this point I think we should normalize the use of sentence ending `.`.  IMHO for consistency let's add it to the end of every field description.

Comment (Huy-Ngo) [reply docs/source/meth/database/comment.rst:17]: I agree, though it often slipped when I write.

**Comment by Huy-Ngo** [comment]
If possible please also look for ambiguous vocabulary that need to be added to glossary.

File: docs/source/appendix/glossary.rst [hunk 3]
```diff
@@ -13,6 +13,12 @@ that are used throughout this report:
       DDoS is the distributed DoS attack, that is, the requests are flooding
       from multiple sources.
 
+   Metadata
+
+      Data that does not describe the information rather than containing
+      the information itself.  This can includes primary key of a table,
```
Comment (McSinyx) [inline docs/source/appendix/glossary.rst:19]: This sentence sounds weird to my ears.

File: docs/source/meth/database/index.rst [hunk 4]
```diff
@@ -1,19 +1,53 @@
...
+group project can have several participating users, we model
+the relation between them as many-to-many.
+We modeled this as double links:
 
-.. TODO write more
+- Each ``User`` object of appropriate role (student or supervisor),
+  there is a field ``projects``.
+- Each ``Project`` has a ``supervisors`` and a ``students`` fields.
+  We model both of these fields as list, since there can be occasions
+  where two supervisors co-supervise a project.
```
Comment (McSinyx) [inline docs/source/meth/database/index.rst:13]: Nothing prevent [more than two](https://en.wikipedia.org/wiki/Two_and_a_Half_Men) supervisors.

**Review by McSinyx — COMMENTED** [review]
Were we to have a glossary, let's define business vocab such as *student*, *supervisor*, *project*, etc.  Not strictly within the scope of this PR though.

**Comment by Huy-Ngo** [comment]
> let's define business vocab such as student, supervisor, project, etc

Yea, I suppose those words need definition too, but I find it hard to define them concretely because they seem too trivial.

File: docs/source/appendix/glossary.rst (hunk 3 above)
Comment (Huy-Ngo) [reply docs/source/appendix/glossary.rst:19]: Hmm it should be `describes the information rather than ...`

**Comment by Huy-Ngo** [comment]
> This can includes

I wish there is a grammar corrector.

**Review by McSinyx — APPROVED** [review]
My requests seemed to be addressed in the recent commit, however there's something wrong with the rebase.

**Comment by Huy-Ngo** [comment]
I just used a spell checker and find some spelling mistake lol.

**Pull request merged**